products/zoningtaxlots/sql/preprocessing.sql

```diff
@@ -25,13 +25,24 @@ RENAME COLUMN wkb_geometry TO geom;
 
 DROP TABLE IF EXISTS dof_dtm_tmp;
 CREATE TABLE dof_dtm_tmp AS (
+    WITH coalesced AS (
+        SELECT
+            bbl,
+            COALESCE(boro::text, LEFT(bbl::text, 1)) AS boro,
+            COALESCE(block::text, SUBSTRING(bbl::text, 2, 5)) AS block,
+            COALESCE(lot::text, SUBSTRING(bbl::text, 7, 4)) AS lot,
+            geom
+        FROM dof_dtm
+    )
+
     SELECT
+        ROW_NUMBER() OVER () AS id,
```

Comment:
Figure at point of grouping of dtm, need to essentially generate pk since bbl isn't necessarily it.